backend/tests/unit/graphql/auth/query_permission_checker/test_super_admin_checker.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from unittest.mock import AsyncMock, MagicMock
+from uuid import uuid4
+
+import pytest
+
+from infrahub.auth import AccountSession, AuthType
+from infrahub.core.constants import AccountRole, GlobalPermissions, InfrahubKind
+from infrahub.core.node import Node
+from infrahub.core.registry import registry
+from infrahub.graphql import GraphqlParams
+from infrahub.graphql.analyzer import InfrahubGraphQLQueryAnalyzer
+from infrahub.graphql.auth.query_permission_checker.interface import CheckerResolution
+from infrahub.graphql.auth.query_permission_checker.super_admin_checker import SuperAdminPermissionChecker
+from infrahub.permissions.local_backend import LocalPermissionBackend
+
+if TYPE_CHECKING:
+    from infrahub.core.branch import Branch
+    from infrahub.core.protocols import CoreAccount
+    from infrahub.database import InfrahubDatabase
+
+
+class PermissionsHelper:
+    def __init__(self) -> None:
+        self._first: None | CoreAccount = None
+        self._second: None | CoreAccount = None
+        self._default_branch: None | Branch = None
+
+    @property
+    def first(self) -> CoreAccount:
+        if self._first:
+            return self._first
+
+        raise NotImplementedError()
+
+    @property
+    def second(self) -> CoreAccount:
+        if self._second:
+            return self._second
+
+        raise NotImplementedError()
+
+    @property
+    def default_branch(self) -> Branch:
+        if self._default_branch:
+            return self._default_branch
```

Comment:
I think this is pretty much a copy of another helper class I did to store states between tests. While not critical for this PR perhaps we should move this to somewhere under tests.helpers.permission instead of redefining what's pretty much the same object. Not super important for me but might remove some extra code throughout the tests?

Reply:
True, I think I'll review this when reworking the default branch checker test that is no longer relevant and needs an update.